packages/react-native-reanimated/src/Colors.ts

```diff
@@ -675,6 +679,23 @@ export function processColorsInProps(props: StyleProps) {
   for (const key in props) {
     if (ColorProperties.includes(key)) {
       props[key] = processColor(props[key]);
+    } else if (
+      NestedColorProperties[key as keyof typeof NestedColorProperties]
+    ) {
+      const nestedPropGroup = props[key] as StyleProps;
+      for (const groupKey in nestedPropGroup) {
+        const nestedProp = nestedPropGroup[groupKey] as StyleProps;
+
+        for (const propName in nestedProp) {
+          if (
+            NestedColorProperties[
+              key as keyof typeof NestedColorProperties
+            ].includes(propName)
+          ) {
+            nestedProp[propName] = processColor(nestedProp[propName]);
+          }
+        }
```

Comment:
Instead of iterating over the entire prop object, let's iterate over `NestedColorProperties` and check if those properties exist in the prop object because the prop object can potentially have more fields than those listed in `NestedColorProperties`

Reply:
prop object being the general object with all the props passed (backgroundColor, width, boxShadow etc.)? Because in current approach I use the loop that was there before (loop through keys in props), and if there is a boxShadow in props (boxShadow being an array by default), I iterate through objects in array (most cases one object), and check for if there is a color prop in boxShadow object

Can you elaborate more? Do you want to move checking the NestedProp outside the parent loop?